From c5b59c459236f3873ec0d7bbb732e0a8533facad Mon Sep 17 00:00:00 2001
From: Adrien Destugues <pulkomandy@pulkomandy.tk>
Date: Sun, 12 Jan 2014 18:36:42 +0100
Subject: Hack to avoid hardcoded library search path.

Makes it possible to find zlib when building as an hybrid package.

includeDir lookup using finddir

Only set includeDir if it's undefined

diff --git a/config.lib b/config.lib
index 8022045..dfaaaa0 100644
--- a/config.lib
+++ b/config.lib
@@ -2636,10 +2636,13 @@ detect_library() {
 			fi
 		fi
 		if [ -z "$res" ] && [ "$os" = "HAIKU" ]; then
-			eval "$2=`ls -1 /boot/common/include/$4*.h 2>/dev/null | egrep \"\/$5\$\"`"
+			if [ -z "$includeDir" ]; then
+				includeDir=`finddir B_SYSTEM_HEADERS_DIRECTORY`
+			fi
+			eval "$2=`ls -1 $includeDir/$4*.h 2>/dev/null | egrep \"\/$5\$\"`"
 			eval "res=\$$2"
 			if [ -z "$res" ]; then
-				log 2 "  trying /boot/common/include/$4$5... no"
+				log 2 "  trying $includeDir/$4$5... no"
 			fi
 		fi
 
-- 
2.2.2

